### PR TITLE
fix: Catch unicodeDecodeError

### LINKED ISSUE #177: Hazelcast repository - Maven issue
https://github.com/jerrinot/hazelcast

```
[[maven]./hazelcast/pom.xml] Traceback (most recent call last):
  File "/usr/local/bin/license-sh", line 2, in <module>
    from license_sh import __main__
  File "/usr/local/lib/python3.8/site-packages/license_sh/__main__.py", line 28, in <module>
    run_license_sh(arguments)
  File "/usr/local/lib/python3.8/site-packages/license_sh/commands/run_license_sh.py", line 91, in run_license_sh
    analyzed_tree = run_analyze(project_to_check, path, dep_tree)
  File "/usr/local/lib/python3.8/site-packages/license_sh/analyze/__init__.py", line 35, in run_analyze
    return analyzer(path, dep_tree)
  File "/usr/local/lib/python3.8/site-packages/license_sh/analyze/maven.py", line 240, in analyze_maven
    licenses_analysis = get_maven_analyze_dict(directory)
  File "/usr/local/lib/python3.8/site-packages/license_sh/analyze/maven.py", line 207, in get_maven_analyze_dict
    license_data = get_analyze_maven_data(directory, dirpath)
  File "/usr/local/lib/python3.8/site-packages/license_sh/analyze/maven.py", line 136, in get_analyze_maven_data
    fetch_maven_licenses(parse_licenses_xml(get_licenses_xml(directory)), license_dir)
  File "/usr/local/lib/python3.8/site-packages/license_sh/analyze/maven.py", line 281, in fetch_maven_licenses
    asyncio.run(fetch_concurrent())
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.8/site-packages/license_sh/analyze/maven.py", line 274, in fetch_concurrent
    output, dep_id = await result
  File "/usr/local/lib/python3.8/asyncio/tasks.py", line 608, in _wait_for_one
    return f.result()  # May raise f.exception().
  File "/usr/local/lib/python3.8/site-packages/license_sh/analyze/maven.py", line 259, in fetch
    return await resp.text(), dep_id
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 1014, in text
    return self._body.decode(encoding, errors=errors)  # type: ignore
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb5 in position 11: invalid start byte

```

### PR DESCRIPTION
**Pull request by S4n60w3n**
Closes #177

### CODE REVIEW

**Review by jvorcak — APPROVED** [review]

File: license_sh/analyze/maven.py [hunk 1]
```diff
@@ -274,6 +274,8 @@ async def fetch_concurrent():
                     output, dep_id = await result
                 except aiohttp.client_exceptions.ClientConnectorCertificateError:
                     continue
+                except UnicodeDecodeError:
```
Comment (jvorcak) [inline license_sh/analyze/maven.py:277]: Can you please elaborate more on why this specific exception (and no others) should be caught?

**Review by jvorcak — CHANGES_REQUESTED** [review]

File: license_sh/analyze/maven.py (hunk 1 above)
Comment (S4n60w3n) [reply license_sh/analyze/maven.py:277]: As much as possible but still we should not disable all

Comment (jvorcak) [reply license_sh/analyze/maven.py:277]: Let's check the mimetype instead of catching an exception please.

**Review by jvorcak — CHANGES_REQUESTED** [review]

File: license_sh/analyze/maven.py (hunk 1 above)
Comment (jankoritak) [reply license_sh/analyze/maven.py:277]: Ideally, also add an explanation comment. Let's avoid cryptic pieces of code.

**Review by jvorcak — APPROVED** [review]
Much better 👍

**Pull request merged**